zoho.go

```diff
@@ -0,0 +1,113 @@
+package zoholab
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Clarilab/zoholab/domain"
+	"github.com/Clarilab/zoholab/middlewares"
+	"github.com/go-resty/resty/v2"
+	"github.com/pkg/errors"
+)
+
+const (
+	reportsUri   = "https://analyticsapi.zoho.eu"
+	validJson    = "true"
+	outputFormat = "JSON"
+	errorFormat  = "JSON"
+	apiVersion   = "1.0"
+	addRow       = "ADDROW"
+)
+
+// ZohoService is the struct for the zoho service.
+type ZohoService struct {
+	restyClient *resty.Client
+}
+
+// NewZohoService instantiates a new zoho service.
+func NewZohoService() *ZohoService {
+	return &ZohoService{
+		restyClient: resty.New(),
+	}
+}
+
+// SetServiceParams sets the params needed to call the zoho api.
+func (s *ZohoService) SetServiceParams(clientID, clientSecret, refreshToken string) {
+	authTokenMiddleware := middlewares.NewAuthTokenMiddleware(clientID, clientSecret, refreshToken)
+
+	s.restyClient.OnBeforeRequest(authTokenMiddleware.AddAuthTokenToRequest)
+}
+
+// GetUri returns the URI for the specified table in the Zoho Analytics account.
+func (s *ZohoService) GetUri(emailID, dbName, tbName string) string {
+	// Join path is not used because the path must not be escaped.
+	return fmt.Sprintf("%s/api/%s/%s/%s", reportsUri, emailID, urlSplCharReplace(dbName), urlSplCharReplace(tbName))
+}
+
+// AddRow adds row to the specified table identified by the URI.
+func (s *ZohoService) AddRow(tableUri string, columnValues map[string]string) (*ZohoAddRowResponse, error) {
+	const errMessage = "could not add row in zoho"
+
+	addedRows := ZohoAddRowResponse{
+		Response: &ZohoResponse{},
+	}
```

Comment:
```suggestion
	var addedRows ZohoAddRowResponse
```

das sollte es schon tun, das wird sowieso von resty befüllt